homeassistant/components/doorbird/config_flow.py

```diff
@@ -0,0 +1,156 @@
+"""Config flow for DoorBird integration."""
+import logging
+import urllib
+
+from doorbirdpy import DoorBird
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import CONF_EVENTS, DOORBIRD_OUI
+from .const import DOMAIN  # pylint:disable=unused-import
+from .util import get_mac_address_from_doorstation_info
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _schema_with_defaults(host=None, name=None):
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST, default=host): str,
+            vol.Required(CONF_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+            vol.Optional(CONF_NAME, default=name): str,
```

Comment:
Should we even ask for a name? Doesn't each device have a name inside doorbird and the user can configure the entity names via entity settings in the UI

Reply:
And devices can be renamed too later.

Reply:
I deleted name and then I put it back because the api doesn't provide a name
https://www.doorbird.com/downloads/api_lan.pdf?rev=0.27

```
{
 "BHA": {
 "RETURNCODE": "1",
 "VERSION": [{
 "FIRMWARE": "000109",
 "BUILD_NUMBER": "15120529",
 "PRIMARY_MAC_ADDR": "1CCAE3700000",
 "RELAYS": ["1", "2", "gggaaa@1", "gggaaa@2"],
 "DEVICE-TYPE": "DoorBird D101"
 }]
 }
}
```

Reply:
But wouldn't it make more sense to ask the user to rename the device via the UI ? Otherwise we would have 2 ways of specifying the name, with one being overridden by the other.